### DISCUSSION TITLE
Whisper crashes with SIGSEGV error

### DISCUSSION THREAD

**Discussion by HristovCodes** [openai/whisper#1630] [Q&A] [answered]
Using whisper with the `--device cuda` option or emitting it because by default it uses cuda throws the following error:

`fish: Job 1, 'whisper pytorch_env/test.mp3 --…' terminated by signal SIGSEGV (Address boundary error)`

I tested it in cpu mode with `--device cpu` and it works flawlessly (it seems) but it's very slow, and when running it even in the console it tells me `UserWarning: Performing inference on CPU when CUDA is available`.

The gpu I'm using is RX 6700 XT if it matters.

Here's the output of `pip list`:
```
Package             Version
------------------- --------------------------
certifi             2022.12.7
charset-normalizer  2.1.1
cmake               3.25.0
ffmpeg-python       0.2.0
filelock            3.9.0
fsspec              2023.4.0
future              0.18.3
idna                3.4
Jinja2              3.1.2
lit                 15.0.7
llvmlite            0.40.1
MarkupSafe          2.1.2
more-itertools      10.1.0
mpmath              1.2.1
networkx            3.0
numba               0.57.1
numpy               1.24.1
openai-whisper      20230314
Pillow              9.3.0
pip                 22.3.1
pytorch-triton-rocm 2.1.0+05d67b9418
regex               2023.8.8
requests            2.28.1
semantic-version    2.10.0
setuptools          65.5.0
setuptools-rust     1.7.0
sympy               1.11.1
tiktoken            0.3.1
torch               2.1.0.dev20230828+rocm5.6
torchaudio          2.1.0.dev20230828+rocm5.6
torchvision         0.16.0.dev20230828+rocm5.6
tqdm                4.66.1
triton              2.0.0
typing_extensions   4.4.0
urllib3             1.26.13
```

Running `torch.cuda.is_available()` also returns True.

I'm running latest version of Arch. And yes I tried using both the stable version of pytorch and the nightly version but both give the same error. I also tried using the `rocm/pytorch` docker image but that also gives the same error. 

Any help will be appreciated because it feels like I've been hitting my head against the wall for days and nothing I try and no ammount of googling seems to help.

**Comment by glangford** [comment, marked as answer]
See this AMD community discussion thread

* https://community.amd.com/t5/infinity-hub-container/radeon-rx6600xt-ubuntu-22-04-and-rocm-woes/m-p/584733

and this whisper discussion (I highlighted a couple of comments within that thread). I am not an AMD user so I can't assist beyond this, but fyi in case it is helpful.

* https://github.com/openai/whisper/discussions/55
* https://github.com/openai/whisper/discussions/55#discussioncomment-3715503
* https://github.com/openai/whisper/discussions/55#discussioncomment-5245204

**Reply by glangford** [reply]
> "In case it helps anyone else, I needed to install rocm-libs and set environment variable HSA_OVERRIDE_GFX_VERSION=10.3.0 to make this work."

**Comment by HristovCodes** [comment]
Setting the correct environment variable seemed to do the job for me.
For anyone else reading this you can do it with the command export like so `export HSA_OVERRIDE_GFX_VERSION=10.3.0`
I did already have amdgpu drivers installed so that migth have been why I didn't need additional configuration other than setting the variable. 
Thanks for pointing me in the rigth direction!
